lib/formatters/getById.js

```diff
@@ -1,23 +1,25 @@
 'use strict';
 
-module.exports = function (rows) {
-  var tags;
+module.exports = function (chal) {
+  var tags = [];
 
   // if no challenge was found
-  if (rows.length === 0) {
-    return {};
+  if (chal.length === 0) {
+    return [];
   }
 
+  if (chal[0].tags_id) {
+    tags = chal.map(function (obj) {
+      return {
+        id: obj.tags_id,
+        name: obj.tags_name
+      };
+    });
+  }
 
-  tags = rows.map(function (obj) {
-    return {
-      id: obj.tags_id,
-      name: obj.tags_name
-    };
-  });
-  delete rows[0].tags_id;
-  delete rows[0].tags_name;
-  rows[0].tags = tags;
+  delete chal[0].tags_id;
```

Comment:
It's not high priority at the moment, so don't get block on this, I'm happy to merge with the current code. But let me know if you have a nice solution :+1: